Update package naming guidelines to encourage consideration of the global namespace (#53514)
JuliaLang/julia: doc/src/tutorials/creating-packages.md

     * `CPLEX.jl` wraps the `CPLEX` library, which can be identified easily in a web search.
     * `MATLAB.jl` provides an interface to call the MATLAB engine from within Julia.

7. Avoid naming a package closely to an existing package
     * `Websocket` is too close to `WebSockets` and can be confusing to users. Rather use a new name such as `SimpleWebsockets`.

> 8. Avoid using a distinctive name that is already in use in another unrelated project.

Author: ```suggestion
8. Avoid using a distinctive name that is already in use in a well known, unrelated project.
```
As Sundar R noted on Zulip, without a "well known" or "popular" clause, names like Oxygen.jl could be suspect as that name is both distinctive and in use elsewhere. However none of the other usages are particularly notable.

     * Don't use the names `Tkinter.jl`, `TkinterGUI.jl`, etc. for a package that is unrelated

>        to the popular `tkinter` python package. Users are likely to assume an affiliation that

Reviewer: > Users are likely to assume an affiliation that does not exist

But this also applies to the case below:

> It's okay to name a package `AlphaZero.jl` ... even without explicit affiliation

so here users would also assume an affiliation that doesn't exist, or?

Author: ```suggestion
       to the popular `tkinter` python package. Users are likely to assume a connection that
```

       does not exist.
     * It's okay to name a package `HTTP.jl` even though it is unrelated to the popular rust

>        crate `http` because "http" is not a distinctive name.

Reviewer 2: Maybe 
```suggestion
       crate `http` because "http" is not a distinctive name, being a more general standard.
```
for some extra clarity on why it isn't a distinctive name might be useful?  Not sure if this is the best wording though.

Reviewer 3: or "because both packages' names refer to the well-known HTTP standard rather than to one another"

Author: I tried to pick a verbiage that highlight what users think the name refers to rather than what the name actually refers to, happy to keep looking for more improvements.

Reviewer 2: That sounds good to me!

     * `CPLEX.jl` wraps the `CPLEX` library, which can be identified easily in a web search.
     * `MATLAB.jl` provides an interface to call the MATLAB engine from within Julia.

7. Avoid naming a package closely to an existing package
     * `Websocket` is too close to `WebSockets` and can be confusing to users. Rather use a new name such as `SimpleWebsockets`.

> 8. Avoid using a distinctive name that is already in use in a well known, unrelated project.

Author: I think it's pretty clear from the first example that this applies to projects from other languages, too. I don't think it's worth adding this text to the summary.

     * Don't use the names `Tkinter.jl`, `TkinterGUI.jl`, etc. for a package that is unrelated

>        to the popular `tkinter` python package. Users are likely to assume a connection that
>        does not exist.

Reviewer 4: ```suggestion
       to the popular `tkinter` python package, even if it provides bindings to Tcl/Tk.
       A package name of `Tkinter.jl` would only be appropriate if the package used Python's
       library to accomplish its work or was spearheaded by the same community of developers.
```

Maybe state when it would be appropriate? In my mind that more directly gets at the connections that a user might assume.

     * It's okay to name a package `HTTP.jl` even though it is unrelated to the popular rust
       crate `http` because in most usages the name "http" refers to the hypertext transfer

>        protocol, not to the `http` python package.

Reviewer 5: ```suggestion
       protocol, not to the `http` rust crate.
```

     * It's okay to name a package `AlphaZero.jl` if it provides an implementation of
       DeepMind's AlphaZero algorithm, even without explicit affiliation with the creators of

>        the AlphaZero algorithm (provided there's no copyright or trademark infringement etc.)

Reviewer 4: It looks like AlphaZero is trademarked.  IANAL and I have no idea how that impacts things like a package, but maybe this isn't the best example?

Reviewer 6: Good point. In fact, I would not use the example of a commercial product since this might suggest that we are encouraging people to violate trademarks. A better example would be an implementation of an academic paper that describes an algorithm - like TSVD.jl.

Author: Though I also don't want to recommend the package name TSVD because it's kind of marginal.

>      * It's okay to name a package `AlphaZero.jl` if it provides an implementation of
>        DeepMind's AlphaZero algorithm, even without explicit affiliation with the creators of
>        the AlphaZero algorithm (provided there's no copyright or trademark infringement etc.)

Author: ```suggestion
     * It's okay to name a package `OpenSSL.jl` if it provides an interface to the OpenSSL
       library, even without explicit affiliation with the creators of the OpenSSL (provided
       there's no copyright or trademark infringement etc.)
```
